### PR TITLE
Find order

### PR DESCRIPTION
**Pull request by adityasharma7**

### CODE REVIEW

File: src/components/OrderFilters.vue [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      <ion-label>{{ $t("Unfillable") }}</ion-label>
+      <ion-checkbox :checked="query.unfillable" @ionChange="updateAppliedFilters($event['detail'].checked, 'unfillable')"/>
+    </ion-item>
+  </ion-list>
+  <ion-list>
+    <ion-list-header>{{ $t("Fulfillment") }}</ion-list-header>
+    <ion-item>
+      <ion-label>{{ $t("Status") }}</ion-label>
+      <ion-select :value="query.status" @ionChange="updateAppliedFilters($event['detail'].value, 'status')" interface="popover">
+        <ion-select-option v-for="status in orderStatusOptions" :key="status" :value="status">{{ status }}</ion-select-option>
```
Comment (adityasharma7) [inline src/components/OrderFilters.vue]: I think value should be id and label should be the description

File: src/components/OrderFilters.vue [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+      </ion-select>
+    </ion-item>
+    <ion-item>
+      <ion-label>{{ $t("Shipping method") }}</ion-label>
+      <ion-select :value="query.shippingMethod" @ionChange="updateAppliedFilters($event['detail'].value, 'shippingMethod')" interface="popover">
+        <ion-select-option v-for="method in shippingMethodOptions" :key="method" :value="method">{{ method }}</ion-select-option>
```
Comment (adityasharma7) [inline src/components/OrderFilters.vue]: I think value should be id and label should be the description

File: src/components/OrderFilters.vue [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+      </ion-select>
+    </ion-item>
+    <ion-item>
+      <ion-label>{{ $t("Ship from location") }}</ion-label>
+      <ion-select :value="query.shipFromLocation" @ionChange="updateAppliedFilters($event['detail'].value, 'shipFromLocation')" interface="popover">
```
Comment (adityasharma7) [inline src/components/OrderFilters.vue:65]: I think here we should use facilityTypeId values

File: src/components/OrderFilters.vue [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  },
+  props: ["poIds", "shippingMethodOptions", "orderStatusOptions"],
+  computed: {
+    ...mapGetters({
+      query: 'order/getOrderQuery'
+    })
+  },
+  methods: {
+    async updateAppliedFilters(value: string, filterName: string) {
+      if (value === this.query[filterName] && !(filterName === 'orderCreated' || filterName === 'promisedDate' || filterName === 'autoCancelDate')) {
```
Comment (adityasharma7) [inline src/components/OrderFilters.vue:133]: I think this should not be needed. Please verify

File: src/store/modules/order/actions.ts [hunk 5]
```diff
@@ -30,6 +41,7 @@ const actions: ActionTree<OrderState, RootState> = {
         const total = resp.data.grouped.orderId.ngroups;
         if (payload.json.params.start && payload.json.params.start > 0) orders = state.list.orders.concat(orders)
         this.dispatch('product/getProductInformation', { orders });
+        this.dispatch('order/fetchStatusChange')
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts]: Why this is needed on every time order is fetch?

File: src/store/modules/order/actions.ts [hunk 6]
```diff
@@ -103,6 +115,86 @@ const actions: ActionTree<OrderState, RootState> = {
...
+  },
+
+  async updateQuery({ state, dispatch }, params) {
+    await dispatch('updateQueryString', params.queryString)
+    const query = prepareOrderQuery({ ...(state.query), poIds: state.poIds, ...params})
+    const resp = await dispatch('findOrders', query)
+    return resp;
+  },
+
+  async updateQueryString({ commit }, queryString) {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts]: I think this is not needed, We could use `updateAppliedFilters`

File: src/store/modules/order/actions.ts [hunk 7]
```diff
@@ -103,6 +115,86 @@ const actions: ActionTree<OrderState, RootState> = {
...
+  },
+
+  async updateAppliedFilters({ state, commit, dispatch }, payload) {
+    if (payload.filterName === 'selectedPoIds') {
+      const poIds = state.query.selectedPoIds;
+      !poIds.includes(payload.value) ? poIds.push(payload.value) : poIds.splice(poIds.indexOf(payload.value), 1)
+      payload.value = poIds
+    }
+    commit(types.ORDER_FILTERS_UPDATED, payload)
+    const resp = await dispatch('updateQuery', { viewSize: process.env.VUE_APP_VIEW_SIZE, viewIndex: 0, queryString: state.query.queryString })
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts]: If queryString is already in state no need to pass it

File: src/store/modules/order/actions.ts [hunk 8]
```diff
@@ -103,6 +115,86 @@ const actions: ActionTree<OrderState, RootState> = {
...
+    commit(types.ORDER_SORT_UPDATED, payload)
+  },
+
+  async updatePoIds({ commit }, payload) {
+    commit(types.ORDER_PO_ID_UPDATED, payload)
+    return payload;
+  },
+
+  async updateQuery({ state, dispatch }, params) {
+    await dispatch('updateQueryString', params.queryString)
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts]: I think we should update query in applied filters and dispatch updateQuery instead

File: src/views/FindOrder.vue [hunk 9]
```diff
@@ -332,11 +278,37 @@ export default defineComponent ({
       ).then(() => {
         event.target.complete();
       })
+    },
+    async openOrderFilterMenu() {
```
Comment (adityasharma7) [inline src/views/FindOrder.vue]:
```suggestion
    async openOrderFilter() {
```

File: src/store/modules/order/actions.ts [hunk 10]
```diff
@@ -103,6 +115,86 @@ const actions: ActionTree<OrderState, RootState> = {
...
+      const poIds = state.query.selectedPoIds;
+      !poIds.includes(payload.value) ? poIds.push(payload.value) : poIds.splice(poIds.indexOf(payload.value), 1)
+      payload.value = poIds
+    }
+    commit(types.ORDER_FILTERS_UPDATED, payload)
+    const resp = await dispatch('updateQuery', { viewSize: process.env.VUE_APP_VIEW_SIZE, viewIndex: 0, queryString: state.query.queryString })
+    return resp;
+  },
+
+  async updateSortOption({ commit }, payload) {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts]:
```suggestion
  async updateSort({ commit }, payload) {
```

File: src/views/FindOrder.vue [hunk 11]
```diff
@@ -272,51 +216,53 @@ export default defineComponent ({
...
+      showOrderItems: true,
+      poIds: {} as any
+    }
+  },
   methods: {
-    async getOrders(vSize?: any, vIndex?: any){
+    async sortOrders(value: string) {
+      this.sort = value
+      await this.store.dispatch('order/updateSortOption', this.sort)
+      this.getOrders();
```
Comment (adityasharma7) [inline src/views/FindOrder.vue]: I think we could this in updateSort action

File: src/views/FindOrder.vue [hunk 12]
```diff
@@ -272,51 +216,53 @@ export default defineComponent ({
...
-          "filter": "docType: ORDER AND orderTypeId: SALES_ORDER"
-        }
-      }
-      if (this.queryString) {
-        payload.json.params.defType = 'edismax'
-        payload.json.params.qf = 'orderId customerPartyName customerPartyId productId internalName'
-        payload.json.params['q.op'] = 'AND'
-        payload.json.query = `*${this.queryString}*`
+
+      const resp = await this.store.dispatch('order/updateQuery', { viewSize, viewIndex, queryString: this.queryString })
```
Comment (adityasharma7) [inline src/views/FindOrder.vue]: If we are already setting the queryString no need to pass it again

Comment (adityasharma7) [inline src/views/FindOrder.vue]: Have separate method for infinite scroll, pass only in that case and skip passing for other cases. updateQuery action should only handle if viewSize is passed with default values

File: src/components/OrderFilters.vue (hunk 2 above)
Comment (ymaheshwari1) [reply src/components/OrderFilters.vue]: Sir, we are fetching the shipping method options from solr facets so we are only getting the id, so used the shipping method getter to display the desc for the shipping method.

File: src/components/OrderFilters.vue (hunk 1 above)
Comment (ymaheshwari1) [reply src/components/OrderFilters.vue]: Sir, in orderStatusOptions we are only getting the id as we are fetching the info from solr facets. So for displaying the label used the env file entry.

File: src/store/modules/order/actions.ts (hunk 5 above)
Comment (ymaheshwari1) [reply src/store/modules/order/actions.ts]: Sir, it's not needed on orders fetch, I just added it for testing, will remove it.

File: src/store/modules/order/actions.ts (hunk 10 above)
Comment (ymaheshwari1) [reply src/store/modules/order/actions.ts]: Updated the method name

File: src/views/FindOrder.vue (hunk 9 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Improved the method name

File: src/store/modules/order/actions.ts (hunk 6 above)
Comment (ymaheshwari1) [reply src/store/modules/order/actions.ts]: Done Sir

File: src/store/modules/order/actions.ts (hunk 7 above)
Comment (ymaheshwari1) [reply src/store/modules/order/actions.ts]: Done Sir

File: src/store/modules/order/actions.ts (hunk 8 above)
Comment (ymaheshwari1) [reply src/store/modules/order/actions.ts]: Done Sir

File: src/views/FindOrder.vue (hunk 11 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Moved the getOrders call to actions.

File: src/views/FindOrder.vue (hunk 12 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Done Sir

Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Improved the code on infinite scroll.

File: src/components/OrderFilters.vue (hunk 3 above)
Comment (ymaheshwari1) [reply src/components/OrderFilters.vue:65]: Sir, used facilityTypeId as a hardcoded value.

File: src/store/modules/order/index.ts [hunk 13]
```diff
@@ -12,7 +12,25 @@ const orderModule: Module<OrderState, RootState> ={
       orders: [],
       total: 0
     },
-    current: {}
+    poIds: {},
+    query: {
+      'status': 'any',
+      'shippingMethod': 'any',
+      'shipFromLocation': 'any',
```
Comment (adityasharma7) [inline src/store/modules/order/index.ts]:
```suggestion
      'shipFromLocation': '',
```

File: src/components/OrderFilters.vue [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+    <ion-item>
+      <ion-label>{{ $t("Shipping method") }}</ion-label>
+      <ion-select :value="query.shippingMethod" @ionChange="updateAppliedFilters($event['detail'].value, 'shippingMethod')" interface="popover">
+        <ion-select-option v-for="method in shippingMethodOptions" :key="method" :value="method">{{ method === 'any' ? 'any' : getShipmentMethodDesc(method) }}</ion-select-option>
+      </ion-select>
+    </ion-item>
+    <ion-item>
+      <ion-label>{{ $t("Ship from location") }}</ion-label>
+      <ion-select :value="query.shipFromLocation" @ionChange="updateAppliedFilters($event['detail'].value, 'shipFromLocation')" interface="popover">
+        <ion-select-option value="any" >{{ $t('any') }}</ion-select-option>
```
Comment (adityasharma7) [inline src/components/OrderFilters.vue]:
```suggestion
        <ion-select-option value="" >{{ $t('any') }}</ion-select-option>
```

File: src/utils/solrHelper.ts [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+
+  if (query.unfillable) {
+    typeFilterSelected.push('_NA_')
+  }
+
+  const typeFilterValues = typeFilterSelected.join(" OR ")
+
+  payload.json.filter = payload.json.filter.concat(` AND facilityId: (${typeFilterValues ? typeFilterValues : '*'})`)
+
+  if (query.shipFromLocation !== 'any') {
```
Comment (adityasharma7) [inline src/utils/solrHelper.ts]:
```suggestion
  if (query.shipFromLocation) {
```

**Pull request merged**